azurelinuxagent/common/errorstate.py

```diff
@@ -0,0 +1,30 @@
+from datetime import datetime, timedelta
+
+ERROR_STATE_DELTA = timedelta(minutes=15)
+
+class ErrorState(object):
+    def __init__(self, min_timedelta):
+        self.min_timedelta = min_timedelta
+
+        self.count = 0;
+        self.timestamp = None
+
+    def incr(self):
+        if self.count == 0:
+            self.timestamp = datetime.utcnow()
+
+        self.count += 1
+
+    def reset(self):
+        self.count = 0
+        self.timestamp = None
+
+    def is_triggered(self):
+        if self.timestamp is None:
+            return False
+
+        delta = datetime.utcnow() - self.timestamp
```

Comment:
This seems to rely on the caller being well behaved. Is it better to have a model where the error state is dependent on the caller or should there be some sort of model where error state manages its own lifecycle? On the one hand I do not want to over-complicate things, and on the other we will miss certain classes of issues (deadlock, e.g.).

Reply:
Discussed this in person, and decided to go with it as it is now.  These are all valid points, but we should be able to learn with what we have. I can make improvements in the future based on our learnings.